plugins/inventory/foreman.py

```diff
@@ -492,87 +539,93 @@ def _populate_report_api(self):
                         except ValueError as e:
                             self.display.warning("Could not set hostvar %s to '%s' for the '%s' host, skipping:  %s" %
                                                  (k, to_native(v), host, to_native(e)))
-            strict = self.get_option('strict')
             hostvars = self.inventory.get_host(host_name).get_vars()
             self._set_composite_vars(self.get_option('compose'), hostvars, host_name, strict)
             self._add_host_to_composed_groups(self.get_option('groups'), hostvars, host_name, strict)
             self._add_host_to_keyed_groups(self.get_option('keyed_groups'), hostvars, host_name, strict)
 
     def _populate_host_api(self):
+        hostnames = self.get_option('hostnames')
+        strict = self.get_option('strict')
         for host in self._get_hosts():
-            if host.get('name'):
```

Comment:
while this looks like a big change in the diff, it's actually just mostly re-indenting the code

Reply:
took me a while to comprehend that ;)